qa/tasks/cephfs/test_volumes.py

```diff
@@ -4345,6 +4345,190 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
+        # tearDown() so that there's zero chance that it interferes with next
+        # test.
+        self.config_set('mgr', self.conf_opt, False)
+
+        # ensure purge threads have no jobs left from previous test so that
+        # next test doesn't have to face unnecessary complications.
+        self._wait_for_trash_empty()
+
+        super().tearDown()
+
+    def _get_sv_path(self, v, sv):
+        sv_path = self.get_ceph_cmd_stdout(f'fs subvolume getpath {v} {sv}')
+        sv_path = sv_path.strip()
+        # delete slash at the beginning of path
+        sv_path = sv_path[1:]
+
+        sv_path = os.path.join(self.mount_a.mountpoint, sv_path)
+        return sv_path
+
+    def _assert_sv_is_absent_in_trash(self, sv, sv_path, sv_files):
+        uuid = self.mount_a.get_shell_stdout('sudo ls volumes/_deleting').\
+            strip()
+
+        trash_sv_path = sv_path.replace('_nogroup', f'_deleting/{uuid}')
+        trash_sv_path = trash_sv_path.replace(sv, '')
+
+        try:
+            sv_files_new = self.mount_a.get_shell_stdout(
+                f'sudo ls {trash_sv_path}')
+        except CommandFailedError as cfe:
+            # in case dir for subvol including files in it are deleted
+            self.assertEqual(cfe.exitstatus, 2)
+            return
+
+        # in case dir for subvol is undeleted yet (but files inside it are).
+        for filename in sv_files:
+            self.assertNotIn(filename, sv_files_new)
+
+    def _assert_trashed_sv_is_unpurged(self, sv, sv_path, sv_files):
+        uuid = self.mount_a.get_shell_stdout('sudo ls volumes/_deleting').\
+                    strip()
+
+        trash_sv_path = sv_path.replace('_nogroup', f'_deleting/{uuid}')
+        trash_sv_path = trash_sv_path.replace(sv, '')
+        sv_files_new = self.mount_a.get_shell_stdout(f'sudo ls {trash_sv_path}').\
+            strip()
+
+        if len(sv_files) == len(sv_files_new):
+            return
+        else:
+            for filename in sv_files:
+                self.assertIn(filename, sv_files_new)
+
+    def test_when_true_subvol_is_trashed_but_stays_unpurged(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging is
+        set to true, running "ceph fs subvolume rm" will move the subvolume
+        to trash but not purge it, that is delete the subvolume from trash.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+
+        self.config_set('mgr', self.conf_opt, True)
+        sv_path = self._get_sv_path(v, sv)
+        self._do_subvolume_io(sv, number_of_files=10)
+        sv_files = self.mount_a.get_shell_stdout(f'sudo ls {sv_path}').\
+            strip().split('\n')
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self._assert_trashed_sv_is_unpurged(sv, sv_path, sv_files)
+        time.sleep(7)
+        self._assert_trashed_sv_is_unpurged(sv, sv_path, sv_files)
```

Comment:
I understand that. My point is you already know the number of files and checking that none of those files have been purged by waiting for 60 seconds is a good enough check.

Reply:
Waiting is redundant if files were deleted at the very beginning of waiting period.

About the point about number of files, I agree.

Reply:
How is it redundant. The config is set before removing the subvolume, so, not a single file should get purged.

Reply:
Ideally, but not in case of bug.

Reply:
In that case, would checking twice ensure that the bug would be caught?

Reply:
This is not making any sense @rishabh-d-dave 

So, you mean:

```
check_something()
sleep(...)
check_something()
```

is better than

```
wait(...)
check_something()
```

especially, when `check_something()` would know the count of the number of files to compare with, just because the former would catch bugs early?

Reply:
If `subvolume rm` will move subvolume to trash and _if we have a bug_, purging would begin immediately even though `pause_purging` is set. In such a case, waiting/sleeping is not needed. `check_something()` (the one before sleep/wait) would fail and cause test to abort.

Reply:
@rishabh-d-dave It seems like you are trying to save a minute or so in test runs, yes? I would prefer simple/straight-froward approaches (especially in test code) in lieu of saving _small_ runtime.

Reply:
Made the change.